src/lib/db/public-signup-token-store.ts

```diff
@@ -31,21 +31,29 @@ interface ITokenUserRow {
     created_at: Date;
 }
 const tokenRowReducer = (acc, tokenRow) => {
-    const { userId, name, ...token } = tokenRow;
+    const { userId, userName, userUsername, roleId, roleName, ...token } =
+        tokenRow;
     if (!acc[tokenRow.secret]) {
         acc[tokenRow.secret] = {
             secret: token.secret,
             name: token.name,
             expiresAt: token.expires_at,
             createdAt: token.created_at,
             createdBy: token.created_by,
-            roleId: token.role_id,
+            role: {
+                id: roleId,
+                name: roleName,
+            },
             users: [],
         };
     }
     const currentToken = acc[tokenRow.secret];
     if (userId) {
-        currentToken.users.push({ userId, name });
+        currentToken.users.push({
+            id: userId,
+            name: userName,
+            username: userUsername,
```

Comment:
Hmm, I can see these two variables being confusing. Would it be better to use `name` and `username`? Or `user` and `username`? I don't know. You've probably thought about this a fair bit yourself, so I'm assuming you've reached the conclusion that `userName` and `userUsername` are the best way to go 💁🏼